packages/components/src/chart/d3chart/chart.js

```diff
@@ -191,16 +165,17 @@ class D3Chart extends Component {
 				className={ classNames( 'd3-chart__container', className ) }
 				style={ { height } }
 			>
+				<div className="d3-chart__tooltip" ref={ this.tooltipRef } />
```

Comment:
Rendering the tooltip below the `<D3Base>` chart was a bit problematic because `<D3Base>` calls `drawChart` when mounted, which requires `this.tooltipRef` to be defined. That produces a race condition and, in some circumstances, `drawChart` would be called before the tooltip ref was set. Making the tooltip not to appear.

Until now, we solved that with a `setTimeout` inside `drawChart`, but I noticed rendering the tooltip above the chart is enough to make sure the ref is set when the chart is mounted. So we could get rid of the `setTimeout` just changing the order we render the chart and the tooltip.

Reply:
Interesting, thanks for the explanation 🙌